### ISSUE TITLE
Howto: Manage client configurations

### ISSUE THREAD

**Issue by jardleex** [HanXHX/ansible-backuppc#5]
Hello @HanXHX,

as already clarified, this module is about handling a BackupPC server installation.
I'm currently trying to build a private role/playbook to configure my backup clients:
- create backuppc group
- create backuppc user
- add public key to authorized_keys
- grant sudo rights on rsync for backuppc user

I am using 'delegate_to' plus 'with_items: "{{ backuppc_hosts }}"'. I choose this attempt to define the hosts only once and not having to maintain a Ansible group and the backuppc_hosts at the same time.

Issues:
As you already pointed out this does not scale very well on larger installations. Another issue I am facing is that Ansible aborts the play when a target is not reachable.

Question: How do you manage your client configurations?

Thanks
Jard

**Comment by HanXHX** [comment]
I use an internal role **on each client**.

There is an extract of the code:

``` yaml
- name: USER | Add specific to user (no root login!)
  user: >
    name="{{ item.ssh_user | default(backup_default_ssh_user) }}"
    shell=/bin/bash
    comment='User for BackupPC'
  with_items: "{{ backup_servers }}"

- name: AUTHORIZED_KEY | Add backkupc's key to backuped user
  authorized_key: >
    user="{{ item.ssh_user | default(backup_default_ssh_user) }}"
    key="{{ item.key }}"
  with_items: "{{ backup_servers }}"

- name: COPY | sudoers
  lineinfile: >
    line='{{ item.ssh_user | default(backup_default_ssh_user) }} ALL=NOPASSWD{{ ":" }} /usr/bin/rsync'
    dest='/etc/sudoers.d/backuppc'
    validate='visudo -cf %s'
    mode=0440
    create=true
  with_items: "{{ backup_servers }}"
```

With vars:

``` yaml
backup_default_ssh_user: 'backuper'
backup_default_ssh_user_home: '/home/{{ backup_default_ssh_user }}'
backup_servers:
  - hostname: 'my.backup.server.com'
    key: 'http://my.backup.server.com/key.pub'
```

On backuppc server, i force using sudo:

``` yaml
backuppc_dest_user: 'backuper'
backuppc_RsyncClientCmd: '$sshPath -q -x -l {{ backuppc_dest_user }} $host sudo $rsyncPath $argList+'
backuppc_RsyncClientRestoreCmd: '$sshPath -q -x -l {{ backuppc_dest_user }} $host sudo $rsyncPath $argList+'
```

**Comment by HanXHX** [comment]
If it's OK for you, you can close this issue.

**Comment by jardleex** [comment]
Hello @HanXHX,

thanks for your reply, that really helped. I build me a custom role with your input.

Best,
Jard
